src/celeritas/optical/detail/OpticalGenAlgorithms.cc

```diff
@@ -8,16 +8,31 @@
 #include "OpticalGenAlgorithms.hh"
 
 #include <algorithm>
+#include <numeric>
 
 #include "corecel/Assert.hh"
 
 namespace celeritas
 {
 namespace detail
 {
+//---------------------------------------------------------------------------//
+struct AccumNumPhotons
+{
+    // Accumulate the number of optical photons from the distribution data
+    CELER_FUNCTION size_type
```

Comment:
```suggestion
    size_type
```
since it's a `.cc`, and you can put it in an anonymous namespace too